packages/engineer/src/feature-dependency-graph.ts

```diff
@@ -0,0 +1,229 @@
+import type { Feature } from '@wixc3/engine-core/src';
+
+export interface Link {
+    source: string;
+    target: string;
+}
+
+export const buildFeatureLinks = (
+    entry: Feature,
+    visitedFeatures: { [propName: string]: number },
+    level: number
+): Array<Link> => {
+    const res = [] as Array<Link>;
+    const deps = entry.dependencies as Array<Feature>;
+    for (const dep of deps) {
+        res.push({ source: entry.id, target: dep.id });
+        if (!(dep.id in visitedFeatures)) {
+            visitedFeatures[dep.id] = level + 1;
+        }
+    }
+    for (const dep of deps) {
+        if (visitedFeatures[dep.id] === level + 1) {
+            res.push(...buildFeatureLinks(dep, visitedFeatures, level + 1));
+        }
+    }
+    return res;
+};
+
+export function template(data: any) {
+    return `<!DOCTYPE html><html>
+<head>
+    <script>
+        graph = ${JSON.stringify(data)};
+    </script>
+    <meta charset="utf-8" />
+    <!-- Load d3.js -->
+    <script src="https://d3js.org/d3.v4.js"></script>
```

Comment:
wait what?
npm?
node_modules?
imports?
types?

Reply:
still in draft - in currently working on it